### PR TITLE
Refactor Wizard, Cards, Map, and Ratings

### PR DESCRIPTION
**Pull request by stevekroh**
Reorganize code with implicit shared hierarchy
 - The add service, update service, add alert pages
 - The tabs of the wizard for those pages
 - Isolate the state needed to render the PointMap in ConnectedPointMap

Various changes to css, including the addition of browserify-css

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
   "presets":  ["es2015", "react"],
+  "plugins": ["transform-object-rest-spread"],
```
Comment (stevekroh) [inline .babelrc]: We can use ES7 object spreads now.

File: .esdoc.json [hunk 2]
```diff
@@ -1,6 +0,0 @@
-{
-  "source": "./www/js",
-	"destination": "./doc",
-  "excludes": ["bundle.js"],
-  "unexportIdentifier": true
-}
```
Comment (stevekroh) [inline .esdoc.json]: We no longer need .esdoc.json. We're using docco instead.

File: .eslintrc [hunk 3]
```diff
@@ -1,35 +1,21 @@
 {
-    "rules": {
-        "indent": [
-            2,
-            2
-        ],
```
Comment (stevekroh) [inline .eslintrc]: Remove indent rules so esformatter and eslint don't conflict.

File: .travis.yml [hunk 4]
```diff
@@ -19,6 +19,7 @@ before_script:
 - npm run build
 script:
 - npm run lint
+- npm run format
```
Comment (stevekroh) [inline .travis.yml]: Re-add the format checks since esformatter and eslint no longer conflict.

**Comment by JRJurman** [comment]
The command `npm run format` fails on windows, the script should reflect the lint syntax; Also, it currently ignores files in the `js` folder, if there is a format ignore, that should include the bundle file.

**Comment by JRJurman** [comment]
`npm run build` fails, because `mkdir` is not a cross platform program, we should use `mkdirp`, a cross-platform npm equivalent for creating directories

File: package.json [hunk 5]
```diff
@@ -3,34 +3,40 @@
...
+    "format:replace": "esformatter -i \"?(src|test)/js/**/*.js\"",
+    "lint": "eslint \"?(src|test)/js/**/*.js\"",
     "livereload": "livereload \"platforms/browser/www/\"",
-    "postinstall": "cordova prepare",
-    "release:android": "cordova build android --release",
+    "postbuild": "cp -r src/index.html src/img www/",
+    "postdebug:android": "node scripts/postdebug-android.js",
+    "postinstall": "cordova prepare --browserify",
     "postrelease:android": "node scripts/postrelease-android.js",
+    "prebuild": "mkdir -p www/",
```
Comment (JRJurman) [inline package.json]: does not work on windows, we should use a cross-platform solution.

**Comment by JRJurman** [comment]
git handles line endings automatically, and right now, `npm run format` fails on a clean branch because of the unix line endings. I vote that we should remove the check, cause it should never be an issue on a sane setup.

On the off-chance that a PR contains an entire file because line endings were changed, we can catch that in a code-review, and revise the setup of the one-off machine.

File: .babelrc (hunk 1 above)
Comment (JRJurman) [inline .babelrc]: will this allow us to remove the Object.assign syntax in the reducers, or should we still be doing that to create _new_ objects.

File: src/css/app.css [hunk 6]
```diff
@@ -1,20 +1,7 @@
...
-
-/* Reset body margin & padding so map touches screen edge */
-body {
-    height: inherit;
-    margin: 0;
-    padding: 0;
-}
-
-#main {
+/*#main {
```
Comment (JRJurman) [inline src/css/app.css]: this should be removed.

File: src/js/app.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  network.monitor();
+
+  const sync = new Sync( database, gateway, store );
+  sync.start();
+
+  gateway.getPoints().then( points => {
+    store.dispatch( reloadPoints( points ) );
+  } );
+
+/*esfmt-ignore-start*/
```
Comment (JRJurman) [inline src/js/app.js]: why are we ignoring format here?

File: src/js/components/ac-drawer.js [hunk 8]
```diff
@@ -5,13 +5,17 @@ import { AppBar, MenuItem, LeftNav, FontIcon } from 'material-ui';
 import { Link } from 'react-router';
 /*eslint-enable no-unused-vars*/
 
+import { connect } from 'react-redux';
```
Comment (JRJurman) [inline src/js/components/ac-drawer.js]: components should not connect to redux... I understand the appeal for doing so in this component, but the action should be passed down from the containers (which do connect redux).

File: src/js/components/block.js [hunk 9]
```diff
@@ -77,24 +138,45 @@ export class FormBlock extends Component {
...
 
+    const {zDepth, className, thinActionButton} = this.props;
+    if ( !isUndefined( zDepth ) ) {
+      props.zDepth = zDepth;
+    }
+    if ( isString( className ) ) {
+      props.className = className;
+    }
+
+    /*eslint-disable no-unused-vars*/
```
Comment (JRJurman) [inline src/js/components/block.js]: why is this here?

File: src/js/components/page.js [hunk 10]
```diff
@@ -2,6 +2,8 @@
 import React, { Component } from 'react';
 /*eslint-enable no-unused-vars*/
 
+import '../../css/page.css';
```
Comment (JRJurman) [inline src/js/components/page.js]: what does this exactly do?

File: src/js/components/rating-selector.js [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+    } );
+
+    // let clearIcon;
+    // if ( !this.props.disabled && this.state.rating != 0 ) {
+    //   clearIcon = (<FontIcon className='material-icons'
+    //                  style={ this.props.style }
+    //                  onClick={ this.onSetRating.bind( this, 0 ) }>
+    //                  clear
+    //                </FontIcon>);
+    // }
```
Comment (JRJurman) [inline src/js/components/rating-selector.js]: The logic here isn't too complicated, if we're not letting users clear ratings, we can just remove this.

File: src/js/components/rating-selector.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+    return (<div style={ { display: 'inline-flex' } }>
+              { stars }
+            </div>);
+  }
+}
+
+RatingSelector.defaultProps = {
+  rating: 3
+};
```
Comment (JRJurman) [inline src/js/components/rating-selector.js]: Shouldn't this value be passed in by the reducer's default value?
Also, I think we should maybe default to 4, as that's deemed as a "good" by most rating systems.. Defaulting to 3 could cause services to have lower ratings than they should...

File: src/js/components/wizard/alert-name-description.js [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        encodingType: navigator.camera.EncodingType.PNG
+      }
+    );
+  }
+
+  getPageFields() {
+    return [ 'name', 'location', 'type' ];
+  }
+
+  getPageContent() {
```
Comment (JRJurman) [inline src/js/components/wizard/alert-name-description.js]: should this be render? why does this component have no render function?

File: .babelrc (hunk 1 above)
Comment (stevekroh) [inline .babelrc]: `Object.assign( {}, { foo: 'bar' }, anotherObject } )` is equivalent to `{ foo: 'bar', ...anotherObject }`

There are some cases where it makes more sense to use the object spread syntax. In the reducers we can keep using Object.assign, or lodash/assign.

Personally, I think we should move to using lodash in the reducers. Lodash has a bunch of useful methods that transform objects but return clones: merge, assign, mergeWith, mergeIn, etc. With Object, we just have Object.assign.

File: src/js/app.js (hunk 7 above)
Comment (stevekroh) [inline src/js/app.js]: This is the largest section of JSX in the entire project. It's cleaner here to have the path and component attributes on the same line. Otherwise, esformatter puts all attributes on their own line. Here, it makes it hard to connect the component with the path.

File: www/.gitignore [hunk 14]
```diff
@@ -0,0 +1,3 @@
+# www/ must exist on checkout, otherwise you get this error:
```
Comment (JRJurman) [inline www/.gitignore]: should this be a .gitkeep file?

File: src/js/components/ac-drawer.js (hunk 8 above)
Comment (stevekroh) [inline src/js/components/ac-drawer.js]: I forgot to move ACDrawer out of components and into containers. Before, Main connected to a bunch of stuff to support ACDrawer's dependencies. When any component called setDrawer or setSnackbar, the *entire* app had to re-render.

File: src/js/components/block.js (hunk 9 above)
Comment (stevekroh) [inline src/js/components/block.js]: Eslint does not yet support a way to detect that <ButtonClass> is a usage of const ButtonClass, triggering the no-unused-vars rule.

The no-unused-vars rule is *extremely* helpful elsewhere, to detect unused imports and props. It makes more sense to use the /*eslint-disable*/ tags in a few places to keep the rule activated everywhere else.

File: src/js/components/page.js (hunk 10 above)
Comment (stevekroh) [inline src/js/components/page.js]: I added browserify-css. Javascript modules import css files, and the browserify module loader injects the css into a `<style>` block. 
Now, we don't have to serve up individual css files, which cuts down on load time.

File: src/js/components/rating-selector.js (hunk 12 above)
Comment (stevekroh) [inline src/js/components/rating-selector.js]: We don't need to hold the intermediate comment state in the store, so I removed the connection to that state slice in the PointCard class. 

I should probably get rid of that reducer.

**Comment by JRJurman** [comment]
the gitignore needs to be updated to include `www/*`

**Comment by JRJurman** [comment]
The app looks shiny and runs smoothly on web, will test on device next and then +1 if everything looks good.

**Comment by JRJurman** [comment]
Adding amenities fails in add-point,
Add-point and add-alert do not show a marker (see below):
image

**Comment by JRJurman** [comment]
:+1:

**Comment by mlw6607** [comment]
The PR of a thousand days begins with a single commit
